### PR TITLE
Qt: Update to version 5.12.3

### PR DESCRIPTION
**Pull request by EchterAgo**
This allows us to properly use color emojis on Linux and we don't need to add libQt5MultimediaGstTools anymore.

We include a workaround to make PyQt5 5.12.3 work with PyInstaller, otherwise it would fail to start saying it can't find QtCore.dll.

See pyinstaller/pyinstaller#4293

We also remove an unneeded copy of the Qt binaries.

### CODE REVIEW

**Comment by EchterAgo** [comment]
Qt 5.12 will even work on Ubuntu 14.04 and Debian 9 when using Ubuntu 16.04 as a base for building the AppImage.

**Comment by EchterAgo** [comment]
This seems to need some more fixes for the Windows build, it can't load `Qt5Core.dll`. In the Windows build the file is located in the root of the installation. Moving `Qt5*.dll` to `PyQt5\Qt\bin` seems to solve that problem.

**Comment by cculianu** [comment]
macOS build on my build VM (El Capital 10.11) fails. It complains that the binaries for Qt are built for macOS 10.12 (Sierra).

I wonder if we should keep a separate `deterministic/requirements-binaries.txt` for Qt 5.11 on macOS. Qt 5.11 was working fine for Emojis on macOS the whole time.

I do not want to update the base requirement for macOS to El Capitan as I believe plenty of people in the world still use El Capitan -- the last time I tried doing (maybe a year ago)  I got lots of user complaints on reddit.

I think we can continue to use 5.11 on macOS for a time. It should work ok -- Qt 5.11 was surprisingly well behaved/bug-free on macOS.

We just have to be careful not to use new Qt 5.11 API calls to respect our macOS 5.11 requirement. 

How do you feel about that?

**Comment by cculianu** [comment]
> This seems to need some more fixes for the Windows build, it can't load `Qt5Core.dll`. In the Windows build the file is located in the root of the installation. Moving `Qt5*.dll` to `PyQt5\Qt\bin` seems to solve that problem.

You may have to play around with `determinstic.spec` in the build-wine directory which is basically a python script that calls into PyInstaller.  You probably need to modify the paths it generates in `a.binaries` ... or something like that. :/ Sucky.

**Comment by EchterAgo** [comment]
Rebased. I think sticking with 5.11 for macOS for now is fine. I'll try to find out what is going on on Windows.

**Comment by EchterAgo** [comment]
The issue is with PyQt5 5.12.3, 5.12.2 is not affected. I'm adding a workaround. See https://github.com/pyinstaller/pyinstaller/issues/4293

**Comment by EchterAgo** [comment]
Ok, I updated this, it now works on Windows.

**Comment by EchterAgo** [comment]
I also added an extra commit to make the Wine build less noisy.

File: gui/qt/__init__.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 try:
     import PyQt5
 except Exception:
-    sys.exit("Error: Could not import PyQt5 on Linux systems, you may try 'sudo apt-get install python3-pyqt5'")
+    sys.exit("Error: Could not import PyQt5, on Linux systems you may try 'sudo apt-get install python3-pyqt5'")
```
Comment (cculianu) [inline gui/qt/__init__.py]: After you commit this, I may go ahead and make this less retarded for Windows and macOS users by doing a conditional on `sys.platform`...

Comment (EchterAgo) [reply gui/qt/__init__.py]: Sure, you can push to the  qt_update_to_5_12 branch

Comment (cculianu) [reply gui/qt/__init__.py]: ok will do so now...

**Comment by cculianu** [comment]
This appears ready. I added the OSX-specific commits on top. Will merge in a minute...

**Comment by EchterAgo** [comment]
```
2bf0c18dc7a1121864bf562145bca436777eb9f15f8f9ede86ff2b9e5fec8590  dist/Electron-Cash-4.0.7-48-g74e490182.exe
484c0ef685131c40575c16636593b4007f691faa0e13deb4ea0d965fa2f62d05  dist/Electron-Cash-4.0.7-48-g74e490182-portable.exe
08c8c49093cbc86a5143d58382c736f97d0ffdd06d48441b418562364d1f9927  dist/Electron-Cash-4.0.7-48-g74e490182-setup.exe
96df852716105b2a99851d413a4c5ce3fc502497aeb1829c48b5c70e93147d9b  dist/Electron-Cash-4.0.7-48-g74e4901-x86_64.AppImage
```

**Pull request merged**